tycho-indexer/src/extractor/dynamic_contract_indexer/cache.rs

```diff
@@ -807,4 +840,112 @@ mod tests {
+            .unwrap();
+        assert_eq!(result.collect::<Vec<_>>(), vec![&100]);
+
+        // Test 3: Key only in one pending layer
+        cache
+            .insert_pending(block2.clone(), "pending_only".to_string(), 200)
+            .unwrap();
+        let result = cache
+            .get_all("pending_only".to_string())
+            .unwrap();
+        assert_eq!(result.collect::<Vec<_>>(), vec![&200]);
+
+        // Test 4: Key in permanent and one pending layer
+        cache.insert_permanent("mixed".to_string(), 300);
+        cache
+            .insert_pending(block1.clone(), "mixed".to_string(), 301)
+            .unwrap();
+        let result = cache
+            .get_all("mixed".to_string())
+            .unwrap();
+        assert_eq!(result.collect::<Vec<_>>(), vec![&301, &300]); // Latest pending first, then permanent
+
+        // Test 5: Key in multiple pending layers and permanent
+        cache.insert_permanent("multi".to_string(), 400);
+        cache
+            .insert_pending(block1.clone(), "multi".to_string(), 401)
+            .unwrap();
+        cache
+            .insert_pending(block2.clone(), "multi".to_string(), 402)
+            .unwrap();
+        cache
+            .insert_pending(block3.clone(), "multi".to_string(), 403)
+            .unwrap();
+        let result = cache
+            .get_all("multi".to_string())
+            .unwrap();
+        assert_eq!(result.collect::<Vec<_>>(), vec![&403, &402, &401, &400]); // Latest to oldest
+
+        // Test 6: Key in some but not all pending layers
+        cache.insert_permanent("sparse".to_string(), 500);
+        cache
+            .insert_pending(block1.clone(), "sparse".to_string(), 501)
+            .unwrap();
+        cache
+            .insert_pending(block3.clone(), "sparse".to_string(), 503)
+            .unwrap();
+        let result = cache
+            .get_all("sparse".to_string())
+            .unwrap();
+        assert_eq!(result.collect::<Vec<_>>(), vec![&503, &501, &500]); // Skips block2, includes others
+
+        // Test 7: Key only in multiple pending layers (no permanent)
+        cache
+            .insert_pending(block1.clone(), "pending_multi".to_string(), 601)
+            .unwrap();
+        cache
+            .insert_pending(block3.clone(), "pending_multi".to_string(), 603)
+            .unwrap();
+        let result = cache
+            .get_all("pending_multi".to_string())
+            .unwrap();
+        assert_eq!(result.collect::<Vec<_>>(), vec![&603, &601]); // Latest to oldest pending only
+
+        // Test 8: After revert, get_all should reflect the new state
+        cache.revert_to(&block2.hash).unwrap();
+        let result = cache
+            .get_all("multi".to_string())
+            .unwrap();
+        assert_eq!(result.collect::<Vec<_>>(), vec![&402, &401, &400]); // Block3 data removed after revert
+
+        // Test 9: After finality, get_all should include finalized data in permanent
```

Comment:
Not going to block here, but I don't like this scheme of packing multiple cases into a single test